src/connector/src/sink/kinesis.rs

```diff
@@ -106,10 +105,8 @@ impl Sink for KinesisSink {
             .stream_name(&self.config.common.stream_name)
             .send()
             .await
-            .map_err(|e| {
-                tracing::warn!("failed to list shards: {}", DisplayErrorContext(&e));
```

Comment:
There should be a principle that "whoever handles the error is responsible for logging the error". The error here is thrown and should be handled (or logged) by the caller.